pkg/provider/aws/resources/s3.go

```diff
@@ -102,14 +103,52 @@ func (bucket *S3Bucket) Configure(params S3BucketConfigureParams) error {
 	return nil
 }
 
-func NewS3Object(bucket *S3Bucket, objectName string, key string, path string) *S3Object {
-	return &S3Object{
-		Name:          objectSanitizer.Apply(fmt.Sprintf("%s-%s", bucket.Name, objectName)),
-		ConstructsRef: bucket.KlothoConstructRef(),
-		Key:           key,
-		FilePath:      path,
-		Bucket:        bucket,
+//
+//func NewS3Object(bucket *S3Bucket, objectName string, key string, path string) *S3Object {
+//	return &S3Object{
+//		Name:          objectSanitizer.Apply(fmt.Sprintf("%s-%s", bucket.Name, objectName)),
+//		ConstructsRef: bucket.KlothoConstructRef(),
+//		Key:           key,
+//		FilePath:      path,
+//		Bucket:        bucket,
+//	}
+//}
+
+type S3ObjectCreateParams struct {
+	AppName  string
+	Ref      core.AnnotationKey
+	Name     string
+	Key      string
+	FilePath string
+}
+
+func (object *S3Object) Create(dag *core.ResourceGraph, params S3ObjectCreateParams) error {
+	bucket, err := core.CreateResource[*S3Bucket](dag, S3BucketCreateParams{
+		AppName: params.AppName,
+		Refs:    core.AnnotationKeySetOf(params.Ref),
+		Name:    params.Ref.ID,
+	})
```

Comment:
you should be able to use the create dependencies call here. Not going to make too big  of a difference since theres only one dep, but i think its a good pattern to use to have centralized logic

Reply:
Hm, can I use it even though the S3Object doesn't yet have its final name (and thus string id)? The S3Object can't determine its name until the S3Bucket gets created and gives itself its name.

Reply:
well so i think we want to try to avoid things like that, i would take in the param of BucketName and Name (not base the name of a reference in case we arent talking in terms of constructs) to generate the object and bucket name

Reply:
this is what we do in the eks scenario to ensure the cluster gets created correctly and the fargate profile or node groups name correlates to the cluster

Reply:
ok